pkg/deploy/template_chart_manager.go

```diff
@@ -0,0 +1,120 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1alpha1"
+	"github.com/kyma-project/lifecycle-manager/pkg/remote"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SKRWebhookTemplateChartManager is a SKRWebhookChartManager implementation that renders
+// the watcher's helm chart and installs it using a native kube-client.
+type SKRWebhookTemplateChartManager struct {
+	config  *SkrChartManagerConfig
+	kcpAddr string
+}
+
+type SkrChartManagerConfig struct {
+	// WebhookChartPath represents the path of the webhook chart
+	// to be installed on SKR clusters upon reconciling kyma CRs.
+	WebhookChartPath       string
+	SkrWebhookMemoryLimits string
+	SkrWebhookCPULimits    string
+	// WatcherLocalTestingEnabled indicates if the chart manager is running in local testing mode
+	WatcherLocalTestingEnabled bool
+	// GatewayHTTPPortMapping indicates the port used to expose the KCP cluster locally for the watcher callbacks
+	GatewayHTTPPortMapping int
+}
+
+func NewSKRWebhookTemplateChartManager(kcpRestConfig *rest.Config, config *SkrChartManagerConfig,
+) (*SKRWebhookTemplateChartManager, error) {
+	resolvedKcpAddr, err := resolveKcpAddr(kcpRestConfig, config)
+	if err != nil {
+		return nil, err
+	}
+	return &SKRWebhookTemplateChartManager{
+		config:  config,
+		kcpAddr: resolvedKcpAddr,
+	}, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) Install(ctx context.Context, kyma *v1alpha1.Kyma) (bool, error) {
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	syncContext := remote.SyncContextFromContext(ctx)
+	chartArgValues, err := generateHelmChartArgs(ctx, syncContext.ControlPlaneClient, kymaObjKey, m.config, m.kcpAddr)
+	if err != nil {
+		return true, err
+	}
+	manifest, err := renderChartToRawManifest(ctx, kymaObjKey, m.config.WebhookChartPath, chartArgValues)
+	if err != nil {
+		return true, err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return true, err
+	}
+	for _, resource := range resources {
+		err := syncContext.RuntimeClient.Patch(ctx, resource, client.Apply, client.ForceOwnership, skrChartFieldOwner)
```

Comment:
would you kindly update this to a concurrent implementation using ErrGroup? otherwise we will have to wait for each resource patch to complete, which develops bottlenecking here